src/domain/app/AppRoute.tsx

```diff
@@ -1,54 +1,60 @@
 import { Helmet } from 'react-helmet-async';
 import { useTranslation } from 'react-i18next';
-import { RouteProps } from 'react-router-dom';
-
-import LoadingSpinner from '../../common/components/spinner/LoadingSpinner';
-import useInstantLogin from '../auth/useInstantLogin';
-import useAuthenticated from '../auth/useAuthenticated';
+import { RouteProps, useNavigate } from 'react-router-dom';
+import { useOidcClient } from 'hds-react';
+import React from 'react';
 
 export type AppRouteProps = Omit<RouteProps, 'lazy' | 'index' | 'children'> & {
   isPrivate?: boolean;
-  title?: string;
+  titleKey?: string;
   noTitle?: boolean;
   // eslint-disable-next-line react/no-unused-prop-types
   index?: true;
 };
 
 function AppRoute({
   isPrivate = false,
-  title,
+  titleKey,
   noTitle,
   element,
 }: AppRouteProps) {
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+
   // Note that instant login should be checked first so that it has
   // access to the path the user attempted to navigate to.
   //
   // For instance, the useAuthenticated hook replaces the current path,
   // in which case the original navigation target would be lost.
-  useInstantLogin(isPrivate);
-  const canRenderComponent = useAuthenticated(isPrivate);
-  const { t } = useTranslation();
+  const { isAuthenticated } = useOidcClient();
 
-  if (!title && !noTitle) {
+  const canRenderComponent = !isPrivate || isAuthenticated();
+
+  if (!titleKey && !noTitle) {
     // eslint-disable-next-line no-console
     console.warn(
       // eslint-disable-next-line max-len
-      'Most app routes should have a title. If a route does not need one, explicitly state is by toggling the noTitle prop.'
+      'Most app routes should have a title. If a route does not need one, explicitly state is by toggling the noTitle prop.',
+      { element }
     );
   }
 
+  const title = titleKey ? `${t(titleKey)} - ${t('appName')}` : t('appName');
```

Comment:
You can leave this as it is if you like, but moving the `const title = ...` right after the `const { t } = useTranslation();` and then using the `title` instead of `titleKey` after that in the code it'd be more like it was before and it would also check whether the `titleKey` resolves to a non-empty string or not.